src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -197,34 +197,43 @@ public void Start()
                 }
 
                 // STEP 2: Open the socket connection to the remote host
-                var cts = new CancellationTokenSource();
-                var tok = cts.Token;
-                
-                tok.Register(() =>
-                {
-                    _client.Close();
-                });
+                var cts = new CancellationTokenSource(ConnectTimeout);
+                cts.CancelAfter(ConnectTimeout);
 
+                using (cts.Token.Register(() => { _client.Close(); }))
+                {
+                    _client.Dispose();
+                }
                 try
                 {
-                    _client.ConnectAsync(_logic.UrlRequest.Host, _logic.UrlRequest.Port).ContinueWith(t =>
+                    using (cts)
                     {
-                        if (!NetworkTaskSuccessful(t)) {
-                            return;
-                        }
+                        _client.ConnectAsync(_logic.UrlRequest.Host, _logic.UrlRequest.Port).ContinueWith(t =>
+                        {
+                            if (!NetworkTaskSuccessful(t))
+                            {
+                                return;
+                            }
 
-                        _queue.DispatchAsync(StartInternal);
+                            _queue.DispatchAsync(StartInternal);
```

Comment:
Make sure your Visual Studio settings are using spaces, not tabs (for consistency and because it shows up weird like this)